awsshell/utils.py

```diff
@@ -118,3 +119,16 @@ def file_contents(self, filename, binary=False):
 
     def file_exists(self, filename):
         return filename in self._file_mapping
+
+
+def force_unicode(obj, encoding='utf8'):
+    """Recursively search lists and dicts for strings to encode as unicode."""
+    if isinstance(obj, dict):
+        for key in obj:
+            obj[key] = force_unicode(obj[key])
+    elif isinstance(obj, list):
+        obj = [force_unicode(val) for val in obj]
+    elif isinstance(obj, six.string_types):
```

Comment:
How is binary handled by the shell? I am not too familiar with how the shell handles binary in terms of: Is binary base64 encoded in responses? If it is not base64 encoded, how do you handle the Python 2 case where the data is binary (i.e. it is considered a ``str``, allowing it to pass the ``six.sting_types`` and ``not six.text_type`` case) but cannot be utf-8 encoded as it is literally binary with no encoding?